errbot/backends/irc.py

```diff
@@ -163,6 +167,15 @@ def join(self, username=None, password=None):
             password = ""
 
         self.connection.join(self.room, key=password)
+        for attempt in range(JOIN_TIMEOUT):
+            time.sleep(.5)
+            if self.joined:
+                break
+            time.sleep(.5)
+        else:
+            self.log.error("Timeout: Could not join %s", self.room)
+            return
```

Comment:
Maybe raise an exception though, instead of silently returning? Some of the other backends raise exceptions too.